### PR TITLE
Add snapshot doc CRC detail

### PR DESCRIPTION
**Pull request by berciuliviu**
Signed-off-by: berciuliviu <lberciu@amazon.com>

# Reason for This PR

Fixes [2068](https://github.com/firecracker-microvm/firecracker/issues/2068)

## Description of Changes

Add details regarding snapshot CRC verification.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: docs/snapshotting/snapshot-support.md [hunk 1]
```diff
@@ -98,7 +98,9 @@ Firecracker is optimized for fast load/resume and it's designed to do some very
 sanity checks only on the vm state file. It only verifies integrity using a 64
 bit CRC value embedded in the vm state file, but this is only as a partial
 measure to protect against accidental corruption, as the disk files and memory
-file need to be secured as well.
+file need to be secured as well. It is important to note that snapshot
+deserialization and CRC computation are done at the same time. Hence,
+the latter can be validated only if the former is successful.
```
Comment (sandreim) [inline docs/snapshotting/snapshot-support.md:103]: I think this has changed. CRC is validated before deserialization.

Comment (berciuliviu) [reply docs/snapshotting/snapshot-support.md:103]: You are correct. I will modify this.

**Comment by berciuliviu** [comment]
@sandreim PTAL

**Review by luminitavoicu — COMMENTED** [review]
Took a look at the entire section. Please remove the duplication of the word `model` on the line 89. It should be
```
The [threat containment model](../design.md#threat-containment) states
```
instead of 
```
The [threat containment model](../design.md#threat-containment) model states
```

**Review by sandreim — APPROVED** [review]

**Review by luminitavoicu — APPROVED** [review]

**Pull request merged**